### ISSUE TITLE
Error Data Structure Malformatted

### ISSUE THREAD

**Issue by 0mars** [mplanchard/falcon-marshmallow#13]
When I call an endpoint with an unknown name I get the json data structure as a string in the description field:
```javascript
{
    "title": "422 Unprocessable Entity",
    "description": "{\"myInvalidFieldName\": [\"Unknown field.\"]}"
}
```
marshmallow==3.2.0
falcon-marshmallow==0.3.0

**Comment by 0mars** [comment]
this is what I did before to put it in a different field:

```
class RequestLoader(Marshmallow):
    def process_resource(self, *args, **kwargs):
        try:
            self.process_resource_inner(*args, **kwargs)
        except ValidationError as err:
            raise HTTPValidationError(status=falcon.status_codes.HTTP_400, errors=err.messages)
        except ValueError as err:
            raise falcon.HTTPError(status=falcon.status_codes.HTTP_400, title='Validation Error', description=str(err))
```
```
{
    "title": "400 Bad Request",
    "errors": {
        "namex": [
            "Unknown field."
        ]
    }
}
```

let me know if you want me to create a PR

falcon-marshmallow==0.2.0
marshmallow==3.0.1

**Comment by mplanchard** [comment]
Hey there! So the behavior you're expecting here is that additional fields are ignored?

I'm thinking this is due to marshmallow 3 and above defaulting to strict mode. Maybe we can add an option to set it to non-strict?

**Comment by 0mars** [comment]
No, the errors are a JSON string inside a JSON string instead of just JSON
all without being treated as a string

**Comment by mplanchard** [comment]
Oh, okay. So you're looking for a nested object, then?

**Comment by 0mars** [comment]
yes, that's what I meant

**Comment by mplanchard** [comment]
This is interesting. The Falcon API [explicitly states](https://falcon.readthedocs.io/en/stable/api/errors.html#base-class) that error descriptions should be strings, which is why, I think, I did things the way they are originally.

It seems like we should be able to override the existing falcon error class and add an `errors` attribute expecting a dictionary, while also properly serializing using `to_json()` and `to_xml()`
